public/app/features/scenes/core/SceneDataTransformer.ts

```diff
@@ -0,0 +1,59 @@
+import { Observable, of, Unsubscribable } from 'rxjs';
+
+import { DataTransformerConfig, LoadingState, PanelData } from '@grafana/data';
+
+import { getTransformationsStream } from '../querying/SceneQueryRunner';
+
+import { SceneObjectBase } from './SceneObjectBase';
+import { sceneGraph } from './sceneGraph';
+import { SceneDataState } from './types';
+
+export interface SceneDataTransformerState extends SceneDataState {
+  transformations?: DataTransformerConfig[];
+}
+
+export class SceneDataTransformer extends SceneObjectBase<SceneDataTransformerState> {
+  private _transformationsSub?: Unsubscribable;
+
+  public activate() {
+    super.activate();
+
+    if (!this.parent || !this.parent.parent) {
+      return;
+    }
+
+    const initialData = sceneGraph.getData(this.parent.parent).state.data;
+
+    if (initialData) {
+      this.transformData(of(initialData));
+    }
+
+    this._subs.add(
+      // Need to subscribe to the parent's parent because the parent has a $data reference to this object
+      sceneGraph.getData(this.parent.parent).subscribeToState({
+        next: (data) => {
+          if (data.data?.state === LoadingState.Done) {
+            this.transformData(of(data.data));
+          }
```

Comment:
I think this also needs to forward loading state, otherwise, that would get lost.

Reply:
Not really, LoadingState is part of the PanelData, and this is exactly what gets passed down to `transformData`. And the transformation processing is returning the entire PanelData object.

Reply:
@dprokop not sure I understand, you are not passing the PanelData with LoadingState.Loading to this.transformData so 
`this.setState({ data });` never get's called with the PanelData with loading state, so how can subscribers to the SceneDataTransformer get it?

Reply:
I think I have misunderstood you @torkelo, opened PR with an update https://github.com/grafana/grafana/pull/60394